### DISCUSSION TITLE
Auto-budget not setting monthly amount

### DISCUSSION THREAD

**Discussion by sa7mon** [firefly-iii/firefly-iii#7316] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I have the following budget:

<img width="1149" alt="Screenshot 2023-04-01 at 3 36 40 PM" src="https://user-images.githubusercontent.com/3712226/229313435-878197fb-8a56-4469-8321-8067ea29651d.png">

It is supposed to set a fixed amount of $50 every month, but when I look at my budgets for this quarter, it shows a budgeted amount of $0

<img width="1113" alt="Screenshot 2023-04-01 at 3 50 34 PM" src="https://user-images.githubusercontent.com/3712226/229313497-42cd4462-6576-45ef-820e-910fdb2c3822.png">

### Debug information

Debug information generated at 2023-04-01 00:00:00 America/Chicago for Firefly III version **6.0.6**.

| Scope | Version |
| --- | --- |
| Firefly III | 6.0.6 |
| Firefly III API | 2.0.1 |
| Build | 441, 01-04-2023 08:04:21 CEST |
| DB version | 19 (exp. 19) |
| PHP | `8.2.4` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | America/Chicago |
| Browser TZ | America/Chicago |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | notice, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Max upload | 67108864 (64 MB) |
| Interface | apache2handler |
| Default language | en_US |
| Default locale | equal |
| BCscale | 12 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Trusted proxies (.env) | `**` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2023-04-01 00:00:00 |
| Session end | 2023-06-30 00:00:00 |
| Session first | 2021-06-08 00:00:00 |
| User ID | 1 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:102.0) Gecko/20100101 Firefox/102.0 |

### Expected behaviour

The budget amount would be automatically set to $50 each month, or $150 quarterly.

### Additional info

Also, the default budgets view shows everything in relation to quarters. Is there any way to see budgets monthly? I'd like to compare month-to-month spending and also see how much I have left this month for each budget.

**Comment by JC5** [comment]
What's the output from the cron job?

**Comment by sa7mon** [comment]
> What's the output from the cron job?

What's the recommended way to check this? I'm running the Docker container option and `crontab` doesn't seem to be installed.

**Comment by sa7mon** [comment]
Ah, found the cron command from other github ssues.

```
root@b1fbbb07db4f:/etc/cron.daily# /usr/local/bin/php /var/www/html/artisan firefly-iii:cron
"Create recurring transactions" cron fired: Recurring transactions cron job fired successfully.
"Create recurring transactions" cron ran with success: Recurring transactions cron job fired successfully.
"Create auto budgets" cron fired: Auto-budget cron job fired successfully.
"Create auto budgets" cron ran with success: Auto-budget cron job fired successfully.
"Send bill warnings" cron fired: Bill warning cron job fired successfully.
"Send bill warnings" cron ran with success: Bill warning cron job fired successfully.
More feedback on the cron jobs can be found in the log files.

root@b1fbbb07db4f:/etc/cron.daily# /usr/local/bin/php /var/www/html/artisan firefly-iii:cron
More feedback on the cron jobs can be found in the log files.
```

I'm not sure "log files" refers to `/var/www/html/storage/logs`, but there are only user login logs there.

**Comment by JC5** [comment]
Lucky you, there's also documentation you can read: https://docs.firefly-iii.org/firefly-iii/advanced-installation/cron/

**Reply by JC5** [reply]
And of course: https://docs.firefly-iii.org/firefly-iii/financial-concepts/organizing/#special-attention

**Reply by sa7mon** [reply]
So if I'm understanding the "trigger dates" correctly, if I'm already past the first of the month (i.e. for April now) and I run the cron job, it will **not** make the budget because we're past the trigger date?

I'll get the cron job setup now, but for this month is there a way to have it create the April budget? Or will have to do a workaround like changing the date inside in the container

**Reply by JC5** [reply]
https://docs.firefly-iii.org/firefly-iii/advanced-installation/cron/#extra-information

**Reply by JC5** [reply]
Or you could set the budgets by hand, that also works.
